src/docfx/lib/git/GitCommitProvider.cs

```diff
@@ -228,45 +228,43 @@ public void Dispose()
             }
         }
 
-        private unsafe (List<Commit>, Dictionary<long, Commit>) LoadCommits(string branchName = null)
+        private unsafe (List<Commit>, Dictionary<long, Commit>) LoadCommits(string committish = null)
         {
+            if (string.IsNullOrEmpty(committish))
+            {
+                committish = "HEAD";
+            }
+
             var commits = new List<Commit>();
             var commitsBySha = new Dictionary<long, Commit>();
 
             // walk commit list
             git_revwalk_new(out var walk, _repo);
             git_revwalk_sorting(walk, 1 << 0 | 1 << 1 /* GIT_SORT_TOPOLOGICAL | GIT_SORT_TIME */);
 
-            if (string.IsNullOrEmpty(branchName))
+            if (git_revparse_single(out var headCommit, _repo, committish) != 0)
             {
-                git_revwalk_push_head(walk);
+                git_object_free(walk);
+                throw Errors.CommittishNotFound(_repoPath, committish).ToException();
             }
-            else
-            {
-                if (git_branch_lookup(out var refBranch, _repo, $"{branchName}", 1 /*locale branch*/) != 0 &&
-                    git_branch_lookup(out refBranch, _repo, $"origin/{branchName}", 2 /*remote branch*/) != 0)
-                {
-                    git_object_free(walk);
-                    throw Errors.GitBranchNotFound(_repoPath, branchName).ToException();
-                }
 
-                var commit = git_reference_target(refBranch);
-                git_revwalk_push(walk, commit);
-                git_object_free(refBranch);
-            }
+            git_revwalk_push(walk, git_object_id(headCommit));
+            git_object_free(headCommit);
 
             while (true)
             {
                 var error = git_revwalk_next(out var commitId, walk);
                 if (error == -31 /* GIT_ITEROVER */)
+                {
                     break;
```

Comment:
should `walk` be freed while exiting as well?

Reply:
it is freed after while loop